src/cargo.rs

```diff
@@ -554,10 +637,13 @@ pub fn rust_flags(config: &Config) -> String {
     }
     handle_llvm_flags(&mut value, config);
     if let Ok(vtemp) = env::var(RUSTFLAGS) {
+        value.push_str(vtemp.as_ref());
         lazy_static! {
             static ref DEBUG_INFO: Regex = Regex::new(r#"\-C\s*debuginfo=\d"#).unwrap();
         }
         value.push_str(&DEBUG_INFO.replace_all(&vtemp, " "));
+    } else {
+        value.push_str(gather_config_rust_flags(config).as_ref());
```

Comment:
Okay so move the lazy_static! block outside of the if statement, and then you can do 

```rust
let vtemp = gather_config_rust_flags(config);
value.push_str(&DEBUG_INFO.replace_all(&vtemp, " "));
```